components/buildList/js/buildList.controller.js

```diff
@@ -6,6 +6,464 @@ buildApp.controller('buildListController', ['$scope', '$q', '$interval', 'buildL
+	 * @param {Object} job object containing information about the job (percentage, status)
+	 * @param {Number} percentage
+	 */
+	function setStatusOnJob(job, percentage) {
+		if (percentage >= 100) {
+			var random = Math.round(Math.random());
+			job.status = (random) ? 'complete' : 'rejected';
+			job.percentage = 100;
+			return;
+		}
+
+		job.status = 'running';
+		job.percentage = percentage; 
+	}
+
+	/**
+	 * Sets the status of a process after analysing the states of its children
+	 * @method setStatusOfProcess
+	 * @param {Array} statesArr states array of the children of the process
+	 */
+	function setStatusOfProcess(statesArr) { 
+		var state;
+
+		if (statesArr.indexOf('running') > -1) {
+			state = 'running';
+		} else if (statesArr.indexOf('rejected') > -1) {
+			state = 'rejected';
+		} else {
+			state = 'complete';
+		}
+
+		return state;
+	}
+
+	/**
+	 * Updates the job state and completeness percentage, based on it's estimated time
+	 * @method updateJob
+	 * @param  {Object} job Job object that is analysed
+	 * @param  {Object} defered
+	 */
+	function updateJob(job, defered) {
+		if (job.status === 'complete' || job.status === 'rejected') {
+			defered.resolve(job.status);
+			return;
+		}
+
+		var estimatedTime = job.estimatedTime * 1000;
+		var percentage = parseInt((timeElapsed * 100) / estimatedTime, 10);
+
+		setStatusOnJob(job, percentage);
+		defered.resolve(job.status);
+	}
+
+	/**
+	 * Updates the firewall status
+	 * @method updateFirewall
+	 * @param  {Object} firewall Firewall that is analysed, for each firewall all it's jobs are requested to be updated
+	 * @param  {Object} firewallDefered
+	 */
+	function updateFirewall(firewall, firewallDefered) {
+		if (firewall.status === 'complete' || firewall.status === 'rejected') {
+			firewallDefered.resolve();
+			return;
+		}
+
+		var jobsDeferals = [],
+			jobsStates = [];
+		firewall.status = 'running';
+
+		angular.forEach(firewall.jobs, function(job) {
+			var jDefered = $q.defer();
+			jobsDeferals.push(jDefered.promise);
+			updateJob(job, jDefered);
+		});
+
+		//update firewall information after all its jobs have been updated
+		$q.all(jobsDeferals).then(function() {
+
+			angular.forEach(firewall.jobs, function(job) {
+				jobsStates.push(job.status);
+			});
+
+			firewall.status = setStatusOfProcess(jobsStates);
+			firewallDefered.resolve(); 
+		});
+	}
+
+	/**
+	 * Updates the build status
+	 * @method updateBuild
+	 * @param  {Object} build Build that is analysed, for each build all its firewalls are requested to be updated
+	 * @param  {Object} firewallDefered
+	 */
+	function updateBuild(build, buildDefered) {
+
+		if (build.status === 'complete' || build.status === 'rejected') {
+			buildDefered.resolve();
+			return;
+		}
+
+		var firewallsPromises = [];
+
+		angular.forEach(build.firewalls, function(firewall) {
+			var firewallDefered = $q.defer();
+			firewallsPromises.push(firewallDefered.promise);
+			updateFirewall(firewall, firewallDefered); 
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove the useless trailing whitespaces at the end of this line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=javascript%3ATrailingWhitespace)